cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/RemoteAgentService.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tether;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpResponse;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+/**
+ * The main class to run the remote agent service.
+ */
+public class RemoteAgentService extends AbstractIdleService {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteAgentService.class);
+  private static final Gson GSON = new Gson();
+  public static final String CONNECT_CONTROL_CHANNEL = "/v3/tethering/controlchannels/";
+
+  private final CConfiguration cConf;
+  private final TetherStore store;
+  private String instanceName;
+  private ScheduledExecutorService executorService;
+
+  @Inject
+  public RemoteAgentService(CConfiguration cConf, TransactionRunner transactionRunner) {
+    this.cConf = cConf;
+    this.store = new TetherStore(transactionRunner);
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    instanceName = cConf.get(Constants.INSTANCE_NAME);
+    int connectionInterval = cConf.getInt(Constants.Tether.CONNECT_INTERVAL,
+                                          Constants.Tether.CONNECT_INTERVAL_DEFAULT);
+    executorService = Executors
+      .newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("tether-control-channel"));
```

Comment:
May be we should use name along the lines of remote agent service?

Reply:
This code was removed after making the change to extend `AbstractScheduledService`